lighthouse-core/gather/gather-runner.js

```diff
@@ -132,12 +133,15 @@ class GatherRunner {
    * Test any error output from the promise, absorbing non-fatal errors and
    * throwing on fatal ones so that run is stopped.
    * @param {!Promise<*>} promise
+   * @param {string=} gathererName
    * @return {!Promise<*>}
    */
-  static recoverOrThrow(promise) {
+  static recoverOrThrow(promise, gathererName) {
     return promise.catch(err => {
       if (err.fatal) {
         throw err;
+      } else {
+        Sentry.captureException(err, {tags: {gatherer: gathererName}, level: 'warning'});
```

Comment:
IMO it's *just* for pulling fatal errors out of the stream. Non-fatal errors are handled on down the line (and due to a continuing lack of `fatal` errors, we should start to consider getting rid of it altogether :)

`collectArtifacts` is explicitly for collating the results of gatherers, including any errors, and has a slot ready made for [the sentry call](https://github.com/GoogleChrome/lighthouse/blob/8fce1465f376eddb88197142b5a2d0c86dc8eb66/lighthouse-core/gather/gather-runner.js#L307)